fe/fe-core/src/main/java/com/starrocks/planner/PlanFragment.java

```diff
@@ -167,7 +171,15 @@ public void setFragmentInPlanTree(PlanNode node) {
      */
     public void setParallelExecNumIfExists() {
         if (ConnectContext.get() != null) {
-            parallelExecNum = ConnectContext.get().getSessionVariable().getParallelExecInstanceNum();
+            int pipelineDop = ConnectContext.get().getSessionVariable().getPipelineDop();
+            int instanceNum = ConnectContext.get().getSessionVariable().getParallelExecInstanceNum();
+            int degreeOfParallelism = ConnectContext.get().getSessionVariable().getDegreeOfParallelism();
+
+            if (ConnectContext.get().getSessionVariable().isEnablePipelineEngine()) {
```

Comment:
Why enable pipeline could change ParallelExecInstanceNum?

Reply:
parallel_fragment_exec_instance_num is not adopted when dop adaption enabled(pipeline_dop==0 && enable_pipeline_engine==true).
when dop adapation enabled,  for a PlanFragment, in default, parallelExecNum is set to half the average number of hardware cores of backends instead of parallel_fragment_exec_instance_num, and pipelineDop is set to 1.